lib/modal_progress_hud.dart

```diff
@@ -24,22 +24,22 @@ import 'package:flutter/material.dart';
 ///
 class ModalProgressHUD extends StatelessWidget {
   final bool inAsyncCall;
-  final double opacity;
-  final Color color;
+  final double? opacity;
+  final Color? color;
   final Widget progressIndicator;
-  final Offset offset;
+  final Offset? offset;
   final bool dismissible;
   final Widget child;
 
   ModalProgressHUD({
-    Key key,
-    @required this.inAsyncCall,
+    Key? key,
+    required this.inAsyncCall,
     this.opacity = 0.3,
     this.color = Colors.grey,
     this.progressIndicator = const CircularProgressIndicator(),
     this.offset,
     this.dismissible = false,
-    @required this.child,
+    required this.child,
   })  : assert(child != null),
```

Comment:
There is no need to `assert(x != null)` anymore when `x` is not a nullable type.